app/views/competitions/_nav.html.erb

```diff
@@ -133,15 +133,16 @@
             icon: "lightbulb",
           }
         end
-        unless @competition.results_posted?
+        if @competition.announced? && !@competition.results_posted?
           if @competition.use_wca_registration?
             nav_items << {
               text: t('.menu.register'),
               path: competition_register_path(@competition),
               icon: "sign in alt"
             }
           end
-          if @competition.registrations.any?
+
+          if !@competition.registration_not_yet_opened? && @competition.has_any_registrations?
```

Comment:
The first part of the `&&` clause uses a `!` on a method that has "not" in its name. So effectively, it's a double negation "If not not yet opened".

Can you look if there's a "if competition registration open" method? Pretty sure there already ought to be one.

Reply:
There is some method to the madness here, because we are only checking if the registration _has opened_, not if it is currently open. Here are the old method definitions: 

```ruby
  def registration_opened?
    use_wca_registration? && !cancelled? && !registration_not_yet_opened? && !registration_past?
  end

  def registration_not_yet_opened?
    registration_open && Time.now < registration_open
  end

  def registration_past?
    registration_close && registration_close < Time.now
  end
```
We can't use the old `registration_opened?` because it invokes `registration_past?`, which returns true if the registration is closed. 

My latest commit changes the logic of `registration_opened?` to ONLY check if the registration has opened in the past, and adds a new `registration_is_open?` method to check if registration is _currently_ open